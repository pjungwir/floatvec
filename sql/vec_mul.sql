-- float mul to scalar lhs
SELECT vec_mul(5::float, floats) FROM measurements WHERE sensor_id IN (4);

-- float mul to scalar rhs
SELECT vec_mul(floats, 5::float) FROM measurements WHERE sensor_id IN (4);

-- float mul to array
SELECT vec_mul(floats, floats) FROM measurements WHERE sensor_id IN (4);

-- float mul to null scalar lhs
SELECT vec_mul(NULL::float, floats) FROM measurements WHERE sensor_id IN (4);

-- float mul to null scalar rhs
SELECT vec_mul(floats, NULL::float) FROM measurements WHERE sensor_id IN (4);

-- float mul scalar to null array lhs
SELECT vec_mul(5::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4);

-- float mul scalar to null array rhs
SELECT vec_mul(NULL::float[], 5::float) FROM measurements WHERE sensor_id IN (4);

-- float mul to null array lhs
SELECT vec_mul(floats, NULL::float[]) FROM measurements WHERE sensor_id IN (4);

-- float mul to null array rhs
SELECT vec_mul(NULL::float[], floats) FROM measurements WHERE sensor_id IN (4);

-- null float mul to null scalar lhs
SELECT vec_mul(NULL::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4);

-- null float mul to null scalar rhs
SELECT vec_mul(NULL::float[], NULL::float) FROM measurements WHERE sensor_id IN (4);

-- null float mul to null array
SELECT vec_mul(NULL::float[], NULL::float[]) FROM measurements WHERE sensor_id IN (4);

-- int mul to scalar lhs
SELECT vec_mul(5::int, ints) FROM measurements WHERE sensor_id IN (4);

-- int mul to scalar rhs
SELECT vec_mul(ints, 5::int) FROM measurements WHERE sensor_id IN (4);

-- int mul to array
SELECT vec_mul(ints, ints) FROM measurements WHERE sensor_id IN (4);

-- int mul to null scalar lhs
SELECT vec_mul(NULL::int, ints) FROM measurements WHERE sensor_id IN (4);

-- int mul to null scalar rhs
SELECT vec_mul(ints, NULL::int) FROM measurements WHERE sensor_id IN (4);

-- int mul scalar to null array lhs
SELECT vec_mul(5::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4);

-- int mul scalar to null array rhs
SELECT vec_mul(NULL::int[], 5::int) FROM measurements WHERE sensor_id IN (4);

-- int mul to null array lhs
SELECT vec_mul(ints, NULL::int[]) FROM measurements WHERE sensor_id IN (4);

-- int mul to null array rhs
SELECT vec_mul(NULL::int[], ints) FROM measurements WHERE sensor_id IN (4);

-- null int mul to null scalar lhs
SELECT vec_mul(NULL::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4);

-- null int mul to null scalar rhs
SELECT vec_mul(NULL::int[], NULL::int) FROM measurements WHERE sensor_id IN (4);

-- null int mul to null array
SELECT vec_mul(NULL::int[], NULL::int[]) FROM measurements WHERE sensor_id IN (4);
