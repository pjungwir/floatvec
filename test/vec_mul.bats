load test_helper

@test "float mul to scalar lhs" {
  result="$(query "SELECT vec_mul(5::float, floats) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{5,NULL,10}" ]
}

@test "float mul to scalar rhs" {
  result="$(query "SELECT vec_mul(floats, 5::float) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{5,NULL,10}" ]
}

@test "float mul to array" {
  result="$(query "SELECT vec_mul(floats, floats) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{1,NULL,4}" ]
}


@test "float mul to null scalar lhs" {
  result="$(query "SELECT vec_mul(NULL::float, floats) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{NULL,NULL,NULL}" ]
}

@test "float mul to null scalar rhs" {
  result="$(query "SELECT vec_mul(floats, NULL::float) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{NULL,NULL,NULL}" ]
}

@test "float mul scalar to null array lhs" {
  result="$(query "SELECT vec_mul(5::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "float mul scalar to null array rhs" {
  result="$(query "SELECT vec_mul(NULL::float[], 5::float) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "float mul to null array lhs" {
  result="$(query "SELECT vec_mul(floats, NULL::float[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "float mul to null array rhs" {
  result="$(query "SELECT vec_mul(NULL::float[], floats) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}


@test "null float mul to null scalar lhs" {
  result="$(query "SELECT vec_mul(NULL::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "null float mul to null scalar rhs" {
  result="$(query "SELECT vec_mul(NULL::float[], NULL::float) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "null float mul to null array" {
  result="$(query "SELECT vec_mul(NULL::float[], NULL::float[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}


@test "int mul to scalar lhs" {
  result="$(query "SELECT vec_mul(5::int, ints) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{5,NULL,10}" ]
}

@test "int mul to scalar rhs" {
  result="$(query "SELECT vec_mul(ints, 5::int) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{5,NULL,10}" ]
}

@test "int mul to array" {
  result="$(query "SELECT vec_mul(ints, ints) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{1,NULL,4}" ]
}


@test "int mul to null scalar lhs" {
  result="$(query "SELECT vec_mul(NULL::int, ints) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{NULL,NULL,NULL}" ]
}

@test "int mul to null scalar rhs" {
  result="$(query "SELECT vec_mul(ints, NULL::int) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "{NULL,NULL,NULL}" ]
}

@test "int mul scalar to null array lhs" {
  result="$(query "SELECT vec_mul(5::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "int mul scalar to null array rhs" {
  result="$(query "SELECT vec_mul(NULL::int[], 5::int) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "int mul to null array lhs" {
  result="$(query "SELECT vec_mul(ints, NULL::int[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "int mul to null array rhs" {
  result="$(query "SELECT vec_mul(NULL::int[], ints) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}


@test "null int mul to null scalar lhs" {
  result="$(query "SELECT vec_mul(NULL::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "null int mul to null scalar rhs" {
  result="$(query "SELECT vec_mul(NULL::int[], NULL::int) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}

@test "null int mul to null array" {
  result="$(query "SELECT vec_mul(NULL::int[], NULL::int[]) FROM measurements WHERE sensor_id IN (4)")";
  echo $result;
  [ "$result" = "NULL" ]
}


