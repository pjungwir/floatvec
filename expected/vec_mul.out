-- float mul to scalar lhs
SELECT vec_mul(5::float, floats) FROM measurements WHERE sensor_id IN (4);
   vec_mul   
-------------
 {5,NULL,10}
(1 row)

-- float mul to scalar rhs
SELECT vec_mul(floats, 5::float) FROM measurements WHERE sensor_id IN (4);
   vec_mul   
-------------
 {5,NULL,10}
(1 row)

-- float mul to array
SELECT vec_mul(floats, floats) FROM measurements WHERE sensor_id IN (4);
  vec_mul   
------------
 {1,NULL,4}
(1 row)

-- float mul to null scalar lhs
SELECT vec_mul(NULL::float, floats) FROM measurements WHERE sensor_id IN (4);
     vec_mul      
------------------
 {NULL,NULL,NULL}
(1 row)

-- float mul to null scalar rhs
SELECT vec_mul(floats, NULL::float) FROM measurements WHERE sensor_id IN (4);
     vec_mul      
------------------
 {NULL,NULL,NULL}
(1 row)

-- float mul scalar to null array lhs
SELECT vec_mul(5::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- float mul scalar to null array rhs
SELECT vec_mul(NULL::float[], 5::float) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- float mul to null array lhs
SELECT vec_mul(floats, NULL::float[]) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- float mul to null array rhs
SELECT vec_mul(NULL::float[], floats) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- null float mul to null scalar lhs
SELECT vec_mul(NULL::float, NULL::float[]) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- null float mul to null scalar rhs
SELECT vec_mul(NULL::float[], NULL::float) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- null float mul to null array
SELECT vec_mul(NULL::float[], NULL::float[]) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- int mul to scalar lhs
SELECT vec_mul(5::int, ints) FROM measurements WHERE sensor_id IN (4);
   vec_mul   
-------------
 {5,NULL,10}
(1 row)

-- int mul to scalar rhs
SELECT vec_mul(ints, 5::int) FROM measurements WHERE sensor_id IN (4);
   vec_mul   
-------------
 {5,NULL,10}
(1 row)

-- int mul to array
SELECT vec_mul(ints, ints) FROM measurements WHERE sensor_id IN (4);
  vec_mul   
------------
 {1,NULL,4}
(1 row)

-- int mul to null scalar lhs
SELECT vec_mul(NULL::int, ints) FROM measurements WHERE sensor_id IN (4);
     vec_mul      
------------------
 {NULL,NULL,NULL}
(1 row)

-- int mul to null scalar rhs
SELECT vec_mul(ints, NULL::int) FROM measurements WHERE sensor_id IN (4);
     vec_mul      
------------------
 {NULL,NULL,NULL}
(1 row)

-- int mul scalar to null array lhs
SELECT vec_mul(5::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- int mul scalar to null array rhs
SELECT vec_mul(NULL::int[], 5::int) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- int mul to null array lhs
SELECT vec_mul(ints, NULL::int[]) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- int mul to null array rhs
SELECT vec_mul(NULL::int[], ints) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- null int mul to null scalar lhs
SELECT vec_mul(NULL::int, NULL::int[]) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- null int mul to null scalar rhs
SELECT vec_mul(NULL::int[], NULL::int) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

-- null int mul to null array
SELECT vec_mul(NULL::int[], NULL::int[]) FROM measurements WHERE sensor_id IN (4);
 vec_mul 
---------
 
(1 row)

